backend/ops_api/ops/resources/budget_line_items.py

```diff
@@ -288,6 +289,36 @@ def __init__(self, model: BaseModel):
         self._response_schema = BudgetLineItemResponseSchema()
         self._response_schema_collection = BudgetLineItemResponseSchema(many=True)
 
+    def associated_with_agreement(self, agreement_id, permission_type: PermissionType) -> bool:
```

Comment:
This method looks the same as `bli_associated_with_agreement` - does that method still exist and if so it would be better to use it rather than duplicating this code.

Reply:
I discussed it with the team and decided to make `check_user_association `. It has the checks that were being repeated within `bli_associated_with_agreement` and `associated_with_agreement`. Please let me know if you have more suggestions. Thank you!